black.py

```diff
@@ -190,6 +191,25 @@ def supports_feature(target_versions: Set[TargetVersion], feature: Feature) -> b
     return all(feature in VERSION_TO_FEATURES[version] for version in target_versions)
 
 
+def merge_exclude(ctx: click.Context, param: click.Parameter, value: str) -> str:
+    """Injects .gitignore in excluded files"""
```

Comment:
As far as I know pythons fnmatch does not implement gits wildmatch. So using the fnmatc is not a good idea, leaves you exposed to a lot of errors. I would suggest something like https://github.com/ambv/retype/pull/15 via parsepath library.

Reply:
@JelleZijlstra @ambv shall we take the implementation from https://github.com/ambv/retype/pull/15 instead then?

Reply:
If it's ok I'll have some time to finish it this week

Reply:
As I said above I don't think using fnmatch is good, we need a proper wildmatch, currently provided via pathspec with recursive ``.gitignore`` discovery. https://github.com/ambv/retype/pull/15/files already does this so we can just take the solution from there and use it here.

Reply:
I totally agree and I was waiting for a reply from @ambv

Reply:
I started a new PR #878 with pathspec support.
Sorry It was much easier than continuing this old one.